File: backend_core_mobile/app/models/notificacion_model.py
# notificacion_model.py
import uuid
from sqlalchemy import Column, String, DateTime, ForeignKey, Boolean
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database.session import Base

class Notificacion(Base):
    __tablename__ = "notificaciones"

    id_notificacion = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    id_usuario = Column(UUID(as_uuid=True), ForeignKey("usuarios.id_usuario"), nullable=False)
    titulo = Column(String(150), nullable=False)
    mensaje = Column(String, nullable=False)
    tipo = Column(String(50), nullable=False)
    leida = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True))

    usuario = relationship("Usuario", foreign_keys=[id_usuario])

File: mobile_app_bcp/lib/shared/models/notificacion_model.dart
// shared/models/notificacion_model.dart
class NotificacionModel {
  final String idNotificacion;
  final String titulo;
  final String mensaje;
  final String tipo;
  final bool leida;
  final DateTime? createdAt;

  NotificacionModel({
    required this.idNotificacion,
    required this.titulo,
    required this.mensaje,
    this.tipo = 'INFORMATIVA',
    this.leida = false,
    this.createdAt,
  });

  factory NotificacionModel.fromJson(Map<String, dynamic> json) {
    return NotificacionModel(
      idNotificacion: json['id_notificacion'] ?? '',
      titulo: json['titulo'] ?? '',
      mensaje: json['mensaje'] ?? '',
      tipo: json['tipo'] ?? 'INFORMATIVA',
      leida: json['leida'] ?? false,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'])
          : null,
    );
  }
}
